moondeck/check/repo_health.py

```
@@ -410,10 +458,22 @@ def render_markdown(new, old):
         for k, v in sorted(new["flash"].items()):
-            L.append(f"| {k} | {_arrow(v, o.get('flash'), k, _kb)} |")
-        L.append("")
+            cap = app_partition_bytes(k) if k.startswith("esp32") else 0
+            cap_s = _kb(cap) if cap else "-"
+            used = f"{(100.0 * v / cap):.0f}%" if cap else "-"
+            built = "yes" if (k in MEASURED_THIS_RUN or k == "desktop") else "carried"
+            L.append(f"| {k} | {_arrow(v, o.get('flash'), k, _kb)} | {cap_s} | {used} | {built} |")
+        L += ["",
+              "`Built: carried` means that firmware was NOT rebuilt this run and its number is "
+              "the previous one, so an absent delta says nothing about the change. `Used` is "
+              "against the app slot in the firmware's own partition table.", ""]
```

Ruff (0.16.2)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)